### PR TITLE
T2f fixes pt3

### PR DESCRIPTION
**Pull request by tejfel**

### CODE REVIEW

File: EquiTrack/t2f/helpers/cost_summary_calculator.py [hunk 1]
```diff
@@ -54,15 +52,18 @@ def get_cost_summary(self):
                   'expenses': expenses}
         return result
 
-    def get_expesnses(self):
+    def get_expenses(self):
         expenses_mapping = defaultdict(Decimal)
-        expenses_qs = self.travel.expenses.select_related('type')
+        expenses_qs = self.travel.expenses.exclude(amount=None).select_related('type')
```
Comment (robertavram) [inline EquiTrack/t2f/helpers/cost_summary_calculator.py]: isn't it better to say amount__isnull=True

File: EquiTrack/t2f/serializers/__init__.py [hunk 2]
```diff
@@ -486,29 +491,33 @@ class Meta:
...
-        return ''
-
-    def get_vision_fi_id(self, obj):
-        return ''
-
-    def to_representation(self, instance):
-        ret = super(InvoiceSerializer, self).to_representation(instance)
-        ret['amount'] = Decimal(ret['amount']).quantize(Decimal('1.'+'0'*instance.currency.decimal_places))
-        return ret
+    def get_amount(self, obj):
```
Comment (ewheeler) [inline EquiTrack/t2f/serializers/__init__.py]: DRY! same method as `InvoiceItemSerializer.get_amount`
add a `TODO` note for the future

File: EquiTrack/t2f/tests/test_invoice_endpoints.py [hunk 3]
```diff
@@ -77,3 +78,50 @@ def test_sorting(self):
         response = self.forced_auth_req('get', reverse('t2f:invoices:list'),
                                         data={'sort_by': 'trip_reference_number'},
                                         user=self.unicef_staff)
+
+    def test_decimal_places(self):
```
Comment (ewheeler) [inline EquiTrack/t2f/tests/test_invoice_endpoints.py]: probably worth adding a test case where `currency.decimal_places` is greater than the precision of `invoice.amount` to be sure that we'll handle `decimal.InvalidOperation` exceptions gracefully

File: EquiTrack/t2f/tests/test_invoice_making.py [hunk 4]
```diff
@@ -19,9 +16,9 @@
 from t2f.tests.factories import CurrencyFactory, ExpenseTypeFactory, WBSFactory, GrantFactory, FundFactory
 
 
-class TravelDetails(APITenantTestCase):
+class InvoiceMaking(APITenantTestCase):
```
Comment (ewheeler) [inline EquiTrack/t2f/tests/test_invoice_making.py]: should we have any tests for negative expense amounts?

File: EquiTrack/EquiTrack/settings/base.py [hunk 5]
```diff
@@ -235,6 +235,8 @@
 
 # See: https://docs.djangoproject.com/en/dev/ref/settings/#use-tz
 USE_TZ = True
+
+USE_INVOICING = os.getenv('USE_INVOICING', False)
```
Comment (ewheeler) [inline EquiTrack/EquiTrack/settings/base.py]: use of invoicing will be the normal default state, so managing this in the future will be less error prone if environment variable and feature gating is inverted to `DISABLE_INVOICING` 

this way in the future if someone forgets to set this flag, the critical invoicing feature won't be mysteriously turned off

File: EquiTrack/t2f/tests/test_travel.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+        profile = self.traveler.profile
+        profile.vendor_number = 'user0001'
+        profile.save()
+
+        country = profile.country
+        country.business_area_code = '0060'
+        country.save()
+
+    def test_cost_summary(self):
```
Comment (ewheeler) [inline EquiTrack/t2f/tests/test_travel.py]: test negative amounts?

File: EquiTrack/t2f/serializers/user_data.py [hunk 7]
```diff
@@ -36,3 +43,23 @@ def get_travel_count(self, obj):
                              Travel.CERTIFIED,
                              Travel.COMPLETED]
         return Travel.objects.filter(traveler=obj).exclude(status__in=excluded_statuses).count()
+
+    def get_business_area(self, obj):
+        workspace = obj.profile.country_override
```
Comment (robertavram) [inline EquiTrack/t2f/serializers/user_data.py]: it should always use profile.country

Comment (tejfel) [inline EquiTrack/t2f/serializers/user_data.py]: So then what's the purpose of country override?

File: EquiTrack/t2f/serializers/__init__.py (hunk 2 above)
Comment (tejfel) [inline EquiTrack/t2f/serializers/__init__.py]: Not exactly the same. The lookup differs a bit. The invoice item has to have an extra `.invoice` in the variable path.

File: EquiTrack/t2f/tests/test_invoice_endpoints.py (hunk 3 above)
Comment (tejfel) [inline EquiTrack/t2f/tests/test_invoice_endpoints.py]: Fixed. Fallback will be the biggest precision possible.

File: EquiTrack/EquiTrack/settings/base.py (hunk 5 above)
Comment (tejfel) [inline EquiTrack/EquiTrack/settings/base.py]: Valid point. Logic inverted.

File: EquiTrack/t2f/tests/test_travel.py (hunk 6 above)
Comment (tejfel) [inline EquiTrack/t2f/tests/test_travel.py]: Expense cannot be negative here.

File: EquiTrack/t2f/tests/test_invoice_making.py (hunk 4 above)
Comment (tejfel) [inline EquiTrack/t2f/tests/test_invoice_making.py]: Expense cannot be negative.

**Pull request merged**